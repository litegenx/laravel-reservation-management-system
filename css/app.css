.router-link-active {
  cursor: pointer;
}

