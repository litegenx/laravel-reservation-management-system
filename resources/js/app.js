require('./bootstrap');
import 'babel-polyfill';
import Vue from 'vue';
import router from './router';
import store from './store';
import App from './components/App';
import vuetify from './plugins/vuetify';

new Vue({
    el: '#app',
    store,
    router,
    components: { App },
    template: '<App/>',
    vuetify,
});
