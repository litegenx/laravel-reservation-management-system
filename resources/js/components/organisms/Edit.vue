<template>
    <v-card>
        <v-form @submit.prevent="save">
            <v-card-title>
                <span class="headline">{{ $t(formTitle) }}</span>
            </v-card-title>

            <v-card-text>
                <v-container grid-list-md>
                    <v-layout
                        v-if="isValid"
                        wrap
                    >
                        <v-flex
                            v-for="form in forms"
                            :key="form.name"
                            xs12
                            sm6
                            md4
                        >
                            <FormItem
                                v-model="formInputs[form.name]"
                                v-validate="form.validate"
                                :form="form"
                                :detail="detail"
                                :form-inputs="formInputs"
                                :data-vv-name="form.text"
                                :validate-errors="errors.collect(form.text)"
                                :increment="increment"
                            />
                        </v-flex>
                    </v-layout>
                </v-container>
            </v-card-text>

            <v-card-actions>
                <v-spacer />
                <v-btn
                    @click="close"
                    color="blue darken-1"
                    text
                >
                    {{ $t('misc.cancel') }}
                </v-btn>
                <v-btn
                    @click="save"
                    color="blue darken-1"
                    text
                >
                    {{ $t('misc.save') }}
                </v-btn>
            </v-card-actions>
        </v-form>
    </v-card>
</template>

<script>
    import { mapGetters, mapActions } from 'vuex';
    import { get, set } from 'lodash';
    import FormItem from '../molecules/FormItem';

    export default {
        components: {
            FormItem,
        },
        props: {
            targetModel: {
                type: String,
                required: true,
            },
            targetId: {
                required: true,
                validator: prop => null === prop || 'number' === typeof prop,
            },
            increment: {
                type: Number,
                required: true,
            },
        },
        data () {
            return {
                formInputs: {},
                nowModel: null,
            };
        },
        computed: {
            ...mapGetters({
                model: 'crud/getTargetModel',
                getModelForms: 'getModelForms',
                getDetailData: 'crud/getDetailData',
            }),
            formTitle () {
                return null === this.targetId ? 'misc.new_item' : 'misc.edit_item';
            },
            forms () {
                return this.getModelForms(this.model);
            },
            detail () {
                return this.targetId ? (this.getDetailData || {}) : {};
            },
            sendInputs () {
                const inputs = {};
                this.forms.map(form => {
                    set(inputs, form.name, this.formInputs[ form.name ]);
                });
                return inputs;
            },
            isValid () {
                return this.targetModel === this.nowModel;
            },
        },
        watch: {
            increment: {
                handler: 'setup',
                immediate: true,
            },
        },
        methods: {
            ...mapActions({
                setModel: 'crud/setModel',
                setDetail: 'crud/setDetail',
                create: 'crud/create',
                edit: 'crud/edit',
            }),
            async setup () {
                this.errors.clear();
                this.clearForm();
                if (!await this.getDetail()) {
                    return;
                }

                this.fillForm();
                this.nowModel = this.targetModel;
            },
            clearForm () {
                this.formInputs = {};
                this.forms.map(form => {
                    this.formInputs[ form.name ] = '';
                });
            },
            async getDetail () {
                await this.setModel(this.targetModel);
                if (this.targetId) {
                    await this.setDetail(this.targetId);
                    if (!this.getDetailData) {
                        this.close();
                        return false;
                    }
                }
                return true;
            },
            fillForm () {
                const inputs = {};
                this.forms.map(form => {
                    inputs[ form.name ] = get(this.detail, form.value, '');
                });
                this.formInputs = inputs;
            },
            update (form, value) {
                this.formInputs[ form.name ] = value;
            },
            save () {
                this.$validator.validateAll().then(async result => {
                    if (!result) {
                        return;
                    }
                    if (this.targetId) {
                        if (await this.edit({ model: this.model, id: this.targetId, data: this.sendInputs })) {
                            this.close();
                        }
                    } else {
                        if (await this.create({ model: this.model, data: this.sendInputs })) {
                            this.close();
                        }
                    }
                });
            },
            close () {
                this.$emit('close-edit');
            },
        },
    };
</script>
