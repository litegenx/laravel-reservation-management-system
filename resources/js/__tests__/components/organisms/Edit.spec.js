import { mount } from '@vue/test-utils';
import Vuex from 'vuex';
import Edit from '../../../components/organisms/Edit';
import app from '../app';
import setupLocalVue from '../setupLocalVue';

jest.useFakeTimers();

describe('Edit', () => {
    it('should render edit', () => {
        const wrapper = mount(app(Edit, 'inner', {
            props: {
                increment: 0,
                targetId: null,
                targetModel: 'test',
            },
        }), setupLocalVue({
            store: new Vuex.Store({
                state: {
                    formInputs: {},
                    nowModel: null,
                },
                actions: {
                    'crud/setModel': () => {
                    },
                },
                getters: {
                    'crud/getTargetModel': () => true,
                    'getModelForms': () => () => [],
                },
            }),
        }));
        expect(wrapper.element).toMatchSnapshot();

        expect(wrapper.findAll('form')).toHaveLength(1);
        expect(wrapper.findAll('button')).toHaveLength(2);
    });
});
