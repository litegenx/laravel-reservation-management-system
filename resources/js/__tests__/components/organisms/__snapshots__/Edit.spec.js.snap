// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit should render edit 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      style="padding: 0px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <div
          class="v-card v-sheet theme--light"
        >
          <form>
            <div
              class="v-card__title"
            >
              <span
                class="headline"
              >
                New Item
              </span>
            </div>
             
            <div
              class="v-card__text"
            >
              <div
                class="container grid-list-md"
              >
                <!---->
              </div>
            </div>
             
            <div
              class="v-card__actions"
            >
              <div
                class="spacer"
              />
               
              <button
                class="v-btn v-btn--flat v-btn--text theme--light v-size--default blue--text text--darken-1"
                type="button"
              >
                <span
                  class="v-btn__content"
                >
                  
                Cancel
            
                </span>
              </button>
               
              <button
                class="v-btn v-btn--flat v-btn--text theme--light v-size--default blue--text text--darken-1"
                type="submit"
              >
                <span
                  class="v-btn__content"
                >
                  
                Save
            
                </span>
              </button>
            </div>
          </form>
        </div>
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
  </div>
</div>
`;
